apps/contract/contracts/BattleBot.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import {ERC721URIStorage} from "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+
+/**
+ * @title BattleBot
+ * @dev Simple NFT implementation for battle bots with metadata storage
+ * Metadata is stored on IPFS and should follow the format:
+ * {
+ *   "version": 1,
+ *   "name": "Battle Bot #1",
+ *   "battlePrompt": "A strategic combat AI that specializes in calculated strikes",
+ *   "image": "ipfs://...",
+ *   "attributes": {
+ *     "attack": 75,
+ *     "defense": 60,
+ *     "speed": 85,
+ *     "mainWeapon": 1
+ *   }
+ * }
+ */
+contract BattleBot is ERC721URIStorage {
```

Comment:
we'd also probably need erc721 mock for testing and local deployment, check erc20mock in repo